activerecord/lib/active_record/model_schema.rb

```diff
@@ -448,6 +471,11 @@ def load_schema
 
         def load_schema!
           @columns_hash = connection.schema_cache.columns_hash(table_name).except(*ignored_columns)
+
+          unless attribute_auto_discovery
+            @columns_hash.reject! { |k| !attributes_to_define_after_schema_loads.key?(k) }
```

Comment:
Why is this filtering necessary? Why can't we just skip the schema load entirely? :confused: